src/main-thread/install.ts

```diff
@@ -37,6 +37,13 @@ export function install(
   baseElement: HTMLElement,
   config: InboundWorkerDOMConfiguration,
 ): Promise<ExportedWorker | null> {
+  const mode = baseElement.dataset['shadowDom'];
+  if (mode != null) {
+    const shadowRoot = baseElement.attachShadow({ mode: (mode || 'open') as ShadowRootMode });
```

Comment:
As written this will support anything truthy.
It may be better to require precisely 'open' or 'closed' so that in the future if new modes are added
we don't need to break compat to support

```suggestion
  if (mode === 'open' || mode === 'closed') {
    const shadowRoot = baseElement.attachShadow({ mode });
```